sled-agent/src/sled_agent.rs

```diff
@@ -1188,6 +1195,71 @@ impl SledAgent {
     pub(crate) fn set_probes(&self, probes: IdHashMap<ProbeCreate>) {
         self.inner.probes.set_probes(probes);
     }
+
+    pub(crate) async fn create_local_storage_dataset(
+        &self,
+        zpool_id: ExternalZpoolUuid,
+        dataset_id: DatasetUuid,
+        request: sled_agent_api::LocalStorageDatasetEnsureRequest,
+    ) -> Result<(), HttpError> {
+        let zpool_name = ZpoolName::External(zpool_id);
```

Comment:
Should this confirm that `zpool_id` is a disk with a local storage dataset our config says is okay to access before attempting to call `Zfs::ensure_dataset()`? The config reconciler exposes a `available_datasets_rx()` watch channel that several sled-agent subsystems consume; we could give `AvailableDatasetReceiver` a `has_local_storage_dataset(&self, zpool_id)` method or something like that?

This would catch two potential problems:

* Disks our config has said we can't use (it would be surprising to get a request from Nexus for such a disk, but it's possible if we're racing, perhaps?)
* Disks our config says we should use, but we haven't been able to ensure the local storage dataset exists (could be a disk problem or a zfs error or ...)